sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -59,6 +61,141 @@ class BucketedReadSuite extends QueryTest with SQLTestUtils with TestHiveSinglet
     }
   }
 
+  // To verify if pruning works, we compare the results before filtering
+  private def checkPrunedAnswers(
+      sourceDataFrame: DataFrame,
+      filterCondition: Column,
+      expectedAnswer: DataFrame): Unit = {
+    val filter = sourceDataFrame.filter(filterCondition).queryExecution.executedPlan
+    assert(
+      filter.isInstanceOf[execution.Filter] ||
+      (filter.isInstanceOf[WholeStageCodegen] &&
```

Comment:
damn forgot about the `WholeStageCodegen` stuff. How about we call `filter.find` to get the underlying relation operator directly?